src/pybind/mgr/dashboard/frontend/cypress/support/commands.ts

```diff
@@ -27,28 +28,57 @@ const fillAuth = () => {
 };
 
 Cypress.Commands.add('login', () => {
-  const username = Cypress.env('LOGIN_USER');
-  const password = Cypress.env('LOGIN_PWD');
-
-  if (auth === undefined) {
-    cy.request({
-      method: 'POST',
-      url: 'api/auth',
-      headers: { Accept: CdHelperClass.cdVersionHeader('1', '0') },
-      body: { username: username, password: password }
-    }).then((resp) => {
-      auth = resp.body;
-      auth.permissions = JSON.stringify(new Permissions(auth.permissions));
-      auth.pwdExpirationDate = String(auth.pwdExpirationDate);
-      auth.pwdUpdateRequired = String(auth.pwdUpdateRequired);
-      auth.sso = String(auth.sso);
-      fillAuth();
-    });
-  } else {
+  requestAuth().then((resp) => {
```

Comment:
@epuertat Isn't it better with the env variables. Atleast in the case of URL for remote origin, that can be anything depending upon the jenkins environment (or the user environment). And rather than changing the code, in this case anyone can go and easily change the `cypress.json` file to modify the URL (both origin and cross-origin one).

Reply:
I usually prefer pure functions (output only depends on input args) or impure functions that expose their internal parameters: `login(user, password)`. If we later want to create a macro/helper wrapper method that gets that user-password fields from the env. variables, that's ok. But the core method should expose its input arguments explicitly.

Reply:
addressed.